File: 5Days-SoftwareArchitecture/typesript/Day2/Composition/Composition.js
// BankAccount class
var BankAccount = /** @class */ (function () {
    function BankAccount() {
    }
    BankAccount.prototype.deposit = function (amount) {
        console.log("Deposited: \u20B9".concat(amount));
    };
    BankAccount.prototype.withdraw = function (amount) {
        console.log("Withdrawn: \u20B9".concat(amount));
    };
    return BankAccount;
}());
// Customer class uses BankAccount (composition)
var Customer = /** @class */ (function () {
    function Customer(account) {
        this.account = account;
    }
    Customer.prototype.performTransactions = function () {
        this.account.deposit(1000);
        this.account.withdraw(500);
        console.log("Transaction completed successfully");
    };
    return Customer;
}());
// Main execution
var account = new BankAccount();
var customer = new Customer(account);
customer.performTransactions();

File: 5Days-SoftwareArchitecture/Java/Day 2/Composition/Composition.java
// BankAccount class
class BankAccount {
    void deposit(double amount) {
        System.out.println("Deposited: " + amount);
    }

    void withdraw(double amount) {
        System.out.println("Withdrawn: " + amount);
    }
}


class Customer {
    private BankAccount account; // Customer has a BankAccount

    Customer(BankAccount account) {
        this.account = account;
    }

    void performTransactions() {
        account.deposit(1000);
        account.withdraw(500);
        System.out.println("Transaction completed successfully");
    }
}


public class Composition {
    public static void main(String[] args) {
        BankAccount acc = new BankAccount(); // create account
        Customer customer = new Customer(acc); // attach to customer

        customer.performTransactions(); // using the account safely
    }
}
